packages/components/src/FilterSelect/FilterSelect.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react"
+import { useButton } from "@react-aria/button"
+import { HiddenSelect, useSelect } from "@react-aria/select"
+import {
+  useSelectState,
+  SelectProps as AriaSelectProps,
+} from "@react-stately/select"
+import { SelectContext, SelectProps, SingleItemType } from "@kaizen/select"
+import { ListBox } from "@kaizen/select/src/Select/components/ListBox"
+import { ListBoxSection } from "@kaizen/select/src/Select/components/ListBoxSection"
+import { ListItem } from "@kaizen/select/src/Select/components/ListItem"
+import { ListItems } from "@kaizen/select/src/Select/components/ListItems"
+import { Option } from "@kaizen/select/src/Select/components/Option"
+import { Overlay } from "@kaizen/select/src/Select/components/Overlay"
+import { SectionDivider } from "@kaizen/select/src/Select/components/SectionDivider"
+import { transformSelectChildren } from "@kaizen/select/src/Select/utils/transformSelectChildren"
+import { Filter, FilterContents, FilterProps } from "~components/Filter"
+import { FilterButtonProps } from "~components/FilterButton"
+import { OverrideClassName } from "~types/OverrideClassName"
+import styles from "./FilterSelect.module.scss"
+
+export type FilterSelectProps = OverrideClassName<
+  Omit<AriaSelectProps<SingleItemType>, "children" | "defaultOpen">
+> & {
+  isOpen: FilterProps["isOpen"]
+  setIsOpen: FilterProps["setIsOpen"]
+  renderTrigger: (triggerButtonProps: FilterButtonProps) => JSX.Element
+  label: string
+  children?: SelectProps["children"]
+  items: AriaSelectProps<SingleItemType>["items"]
+}
+
+export const FilterSelect = ({
+  isOpen,
+  setIsOpen,
+  renderTrigger,
+  label,
+  children,
+  ...restProps
+}: FilterSelectProps): JSX.Element => {
+  // Ref will be populated by Filter
+  const [triggerRef, setTriggerRef] = useState<
+    React.RefObject<HTMLButtonElement>
+  >({ current: null })
+
+  const ariaSelectProps: AriaSelectProps<SingleItemType> = {
+    label,
+    children: transformSelectChildren,
+    isOpen,
+    onOpenChange: setIsOpen,
+    ...restProps,
+  }
+
+  const state = useSelectState(ariaSelectProps)
+
+  const { triggerProps, menuProps } = useSelect(
+    ariaSelectProps,
+    state,
+    triggerRef
+  )
+
+  const { buttonProps } = useButton(triggerProps, triggerRef)
+
+  const items = Array.from(state.collection)
+
+  return (
+    <>
+      <HiddenSelect label={label} state={state} triggerRef={triggerRef} />
+      <Filter
+        isOpen={state.isOpen}
+        setIsOpen={state.setOpen}
+        renderTrigger={(): JSX.Element =>
+          renderTrigger({
+            selectedValue: state.selectedItem?.textValue || undefined,
+            label,
+            isOpen,
+            ...buttonProps,
+          })
+        }
+        onTriggerLoaded={setTriggerRef}
```

Comment:
Introducing API for `react-aria` 😢

Reply:
Yeah I really hated that but unfortunately all the hooks rely on each other too much and we couldn't get away with not doing it 😭

Reply:
What do you think about adding something to signify this is a specific prop for react aria?
eg. `setReactAriaTriggerRef`

Or even introduce a `reactAria` prop that accepts this?

Reply:
Nah I don't want to be too tied into saying a prop only exists for a library. If some other library decides it also needs a ref, it can use it too.
That being said, if we want to call it something more specifically related to having a `ref` that's fine too. I also considered `onMount` lol, cos that's what it's doing.

Reply:
Ah yeah, bring back ol' class terms, `onTriggerMount`

Reply:
Actually, just being wary, but we're stacking a lot of props for the trigger. Should we consider an object prop for `trigger` to encapsulate it a bit?

```
trigger: {
   render: () => {},
   onMount: () => {}
}
```

Reply:
The mounting is an extremely unlikely case tbh. Adding another layer to the trigger prop is probably not very nice. Also we're back at the confusing `trigger` prop naming 😂 
I think if we keep it as `onMount` it could be more extensible moving forward if we want more things later since it's attached to an agnostic event as opposed to tied to an event for an element.

Reply:
Hmm I suppose at this stage, all the props are technically for the trigger lol